src/libDirectoryService/DSBlockPostProcessing.cpp

```diff
@@ -384,6 +383,35 @@ void DirectoryService::ProcessDSBlockConsensusWhenDone(
 
     {
         lock_guard<mutex> g(m_mutexAllPOW1);
+
+        if (m_mode != IDLE)
+        {
+            //Copy POW1 to POW2
+            lock_guard<mutex> g2(m_mutexAllPOW2);
+            lock_guard<mutex> g3(m_mutexAllPoWConns);
+            m_allPoW2s.clear();
+
+            for (auto i : m_allPoW1s)
```

Comment:
Some suggestions of using auto-for
- use `auto i` if you want a copy
- use `auto &i` if you want to modify the item
- use `auto const &i` if you want a read-only reference

Here `auto const &i` will do the job.

Reply:
ok